### PR TITLE
Validate file names / mime-type in upload endpoint with wael

### PR DESCRIPTION
**Pull request by hadeel43**

### CODE REVIEW

File: api/src/app.js [hunk 1]
```diff
@@ -23,30 +41,65 @@ const FileUpload = upload.fields(
...
-        res.status(500).json({ errors });
-    } else{
+        res.status(500).json({
+            errors
+        });
+    } else {
         Apply(req, res)
     }
 });
+const allowedFileTypes = ['jpeg', 'jpg', 'png', 'gif'];
```
Comment (M3kH) [inline api/src/app.js]: Not needed :-) use the `fileType` function :-)

Comment (isaachinman) [reply api/src/app.js]: Can you explain what you mean? I believe this approach is fine, although I'd like to see real mime types, not these extension strings. The point is that the file types accepted for this CV/ML upload endpoint are a subset of the total filetypes accepted by our `multer` instance.

File: api/src/app.js [hunk 2]
```diff
@@ -23,30 +41,65 @@ const FileUpload = upload.fields(
...
-app.post('/upload', FileUpload, (req, res) => Upload(req, res));
+app.post('/upload', FileUpload, (req, res) => {
+
+    fieldName.forEach((element) => {
+        if (file.fieldname === element) {
+            fileType(file, cb)
+            Upload(req, res)
+        }
+    });
+    if (file.fieldname === 'input_ file_assignment') {
```
[the remark this answers is not in the document]
Comment (isaachinman) [reply api/src/app.js]: Also just noticed, you have no `else` on this condition. We need to return HTTP responses in all possible cases.

File: api/src/app.js (hunk 1 above)
Comment (M3kH) [reply api/src/app.js]: I mean this a duplication of what is declared inside the `fileType` function.

File: api/src/app.js [hunk 3]
```diff
@@ -1,52 +1,136 @@
...
+
+function fileType(file, cb) {
+    const fileTypeAccepted = allowedMimeTypes.includes(file.mimetype);
+    const fieldNameAccepted = fields.includes(file.fieldname);
+    if (fieldNameAccepted) {
+        if (fileTypeAccepted) {
+            return cb(null, true);
+        }
+
+        cb(
```
Comment (isaachinman) [inline api/src/app.js]: This function needs to return consistently. Currently, you're only returning in your "positive" case, but not in your negative cases. I believe if a mime type is rejected, you're going to end up firing both `cb` calls here (line 57 and line 64).

Please double check your error handling, and make sure to actually test failures!

**Comment by M3kH** [comment]
Please update the branch to the latest `master`.

**Comment by isaachinman** [comment]
Just a heads up - commit messages like `update app.js` and `update branch` are not appropriate and should probably be refused in the future. Please accurately describe the work in each commit.

**Pull request merged**